src/SavedSearch.php

```diff
@@ -874,7 +874,37 @@ public function getMine(string $itemtype = null, bool $inverse = false, bool $en
             $searches[$data['id']] = $data;
         }
 
-        return $searches;
+        // get personal order
+        $user               = new User();
+        $personalorderfield = $this->getPersonalOrderField();
+        $ordered            = [];
+
+        $personalorder = [];
+        if ($user->getFromDB(Session::getLoginUserID())) {
+            $personalorder = importArrayFromDB($user->fields[$personalorderfield]);
+        }
+        if (!is_array($personalorder)) {
+            $personalorder = [];
+        }
+
+        // Add on personal order
+        if (count($personalorder)) {
+            foreach ($personalorder as $val) {
+                if (isset($searches[$val])) {
+                    $ordered[$val] = $searches[$val];
+                    unset($searches[$val]);
```

Comment:
To make developper aware that `$personalorder` is a set of saved searches IDs.
```suggestion
            foreach ($personalorder as $id) {
                if (isset($searches[$id])) {
                    $ordered[$id] = $searches[$id];
                    unset($searches[$id]);
```